pkg/services/ngalert/metrics/historian.go

```diff
@@ -0,0 +1,51 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type Historian struct {
+	TransitionsTotal      *prometheus.CounterVec
+	TransitionsFailed     *prometheus.CounterVec
+	WritesTotal           prometheus.Counter
+	WritesFailed          prometheus.Counter
+	ActiveWriteGoroutines prometheus.Gauge
+	PersistDuration       prometheus.Histogram
+}
+
+func NewHistorianMetrics(r prometheus.Registerer) *Historian {
+	return &Historian{
+		TransitionsTotal: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "state_history_transitions_total",
+			Help:      "The total number of state transitions processed by the state historian.",
+		}, []string{"org"}),
+		TransitionsFailed: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "state_history_transitions_failed_total",
+			Help:      "The total number of state transitions that failed to be written.",
+		}, []string{"org"}),
+		WritesTotal: promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "state_history_batch_writes_total",
+			Help:      "The total number of state history batches that were attempted to be written.",
+		}),
+		WritesFailed: promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "state_history_batch_writes_failed_total",
+			Help:      "The total number of failed writes of state history batches.",
+		}),
```

Comment:
Apologies if my comment was unclear, but after the changes, the ones that should include the word "batch" are `TransitionsTotal` and `TransitionsFailed`.

So:

`state_history_transitions_batch_failed_total` and `state_history_transitions_batch_total`

The write metric it's just the single write operation, as there's nothing to count, unlike the transitions.

Reply:
But `TransitionsTotal` also is not counting batches. It's counting number of individual records. If 3 batches are written, each with 10 rows, the value is 30, it's the number of actual records we expect to see in the backend.

Writes is actually what corresponds to batches. You get one write attempt per batch.

Maybe we should just drop the `_batch_` segment altogether? It feels misleading in general.

Reply:
I tried this, LMK what you think.